config/v3_4_experimental/translate/translate.go

```diff
@@ -27,9 +28,33 @@ func translateIgnition(old old_types.Ignition) (ret types.Ignition) {
 	return
 }
 
+func translateFileEmbedded1(old old_types.FileEmbedded1) (ret types.FileEmbedded1) {
+	tr := translate.NewTranslator()
+	tr.Translate(&old.Append, &ret.Append)
+	tr.Translate(&old.Contents, &ret.Contents)
+	if util.CheckForSpecialModeBits(old.Mode) {
+		ret.Mode = util.IntToPtr(*old.Mode & 0777)
```

Comment:
To be more explicit about our intent, I'd be inclined to mask out only the bits we actually care about: `&^ 07000`

Also, this is an unusual thing to do during translation, so let's add a really good comment here.